collect_app/src/main/java/org/odk/collect/android/utilities/WidgetAppearanceUtils.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2019 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.odk.collect.android.utilities;
+
+import android.content.Context;
+import android.content.res.Configuration;
+
+import org.javarosa.form.api.FormEntryPrompt;
+import org.odk.collect.android.external.ExternalDataUtil;
+
+import java.util.Locale;
+
+import androidx.annotation.NonNull;
+import timber.log.Timber;
+
+public class WidgetAppearanceUtils {
+    private static final String EXCEPTION_PARSING_COLUMNS = "Exception parsing columns";
+
+    // Date appearances
```

Comment:
I wish we'd gone for one or more enums (`DateAppearance`, `SelectAppearance`, etc) here instead of a bunch of constants. Something to maybe come back to at some point. It would also be good to mark the appearances that have been renamed as deprecated (e.g. `compact`, `selfie`).